### ISSUE TITLE
Clear change and improve Xgboost code

### ISSUE THREAD

**Issue by benjamindeneu** [maximiliense/Data-science-2.0#11]
change to fit the standards of others types of models (sklearn torch).

- [x] separate load/fit/save

- [x] add possibility to chose cpu instead of gpu

- [x] number of labels as parameters
